src/browser/api/window.js

```diff
@@ -440,6 +440,22 @@ Window.create = function(id, opts) {
         ofEvents.emit(route.window('openfin-diagnostic/unload', uuid, name, true), url);
     };
 
+    // Hack: closing a window before content is finish loading in 10.66.40.* causes the renderer to crash.
+    // Hide the window instead, and close from the core when it's safe.
+    let wasCloseRequested = false;
+    const preventClose = (event) => {
+        wasCloseRequested = true;
+        browserWindow.hide();
+    };
+    ofEvents.on(route.window('close-requested', uuid, name), preventClose);
+    ofEvents.once(route.window('initialized', uuid, name), () => {
+        ofEvents.removeListener(route.window('close-requested', uuid, name), preventClose);
+        if (wasCloseRequested) {
+            Window.close({ uuid, name });
```

Comment:
This should force close here. under normal circumstances any close-requested event would not fire (assuming the user closes the window before the context is ready).

Reply:
This would need to ensure that downstream `will-close` events fire then

Reply:
Talked offline, let's ignore that one.